src/client/metering.rs

```diff
@@ -0,0 +1,201 @@
+                subject_id: feature_id,
+                entity_id: entity_id,
+                segment_id: segment_id,
+            }))
+            .map_err(|_| crate::errors::Error::MeteringError {})
+    }
+
+    /// Record the evaluation of a property, for eventual transmission to the server.
+    pub fn record_property_evaluation(
+        &self,
+        property_id: String,
+        entity_id: String,
+        segment_id: Option<String>,
+    ) -> crate::errors::Result<()> {
+        self.evaluation_event_sender
+            .send(EvaluationEvent::Property(EvaluationEventData {
+                subject_id: property_id,
+                entity_id: entity_id,
+                segment_id: segment_id,
+            }))
+            .map_err(|_| crate::errors::Error::MeteringError {})
+    }
+}
+
+pub(crate) struct MeteringTask {
+    _thread_handle: crate::utils::ThreadHandle<()>,
+}
+
+pub(crate) struct EvaluationEventData {
+    /// ID if the subject being evaluated. E.g. feature ID.
+    pub subject_id: String,
+    /// The ID of the Entity against which the subject was evaluated.
+    pub entity_id: String,
+    /// If applicable, the segment the subject was associated to during evaluation.
+    pub segment_id: Option<String>,
+}
+
+pub(crate) enum EvaluationEvent {
+    Feature(EvaluationEventData),
+    Property(EvaluationEventData),
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::models::ConfigurationJson;
+    use crate::models::MeteringDataJson;
+    use crate::network::http_client::WebsocketReader;
+    use crate::NetworkResult;
+
+    struct ServerClientMock {
+        metering_data_sender: mpsc::Sender<()>,
+    }
+    struct WebsocketMockReader {}
+    impl WebsocketReader for WebsocketMockReader {
+        fn read_msg(&mut self) -> tungstenite::error::Result<tungstenite::Message> {
+            unreachable!()
+        }
+    }
+
+    impl ServerClientMock {
+        fn new() -> (ServerClientMock, mpsc::Receiver<()>) {
+            let (sender, receiver) = mpsc::channel();
+            (
+                ServerClientMock {
+                    metering_data_sender: sender,
+                },
+                receiver,
+            )
+        }
+    }
+
+    impl ServerClient for ServerClientMock {
+        #[allow(unreachable_code)]
+        fn get_configuration(
+            &self,
+            _configuration_id: &ConfigurationId,
+        ) -> NetworkResult<ConfigurationJson> {
+            unreachable!()
+        }
+
+        #[allow(unreachable_code)]
+        fn get_configuration_monitoring_websocket(
+            &self,
+            _collection: &ConfigurationId,
+        ) -> NetworkResult<impl WebsocketReader> {
+            unreachable!() as crate::NetworkResult<WebsocketMockReader>
+        }
+
+        fn push_metering_data(&self, _data: &MeteringDataJson) -> NetworkResult<()> {
+            self.metering_data_sender.send(()).unwrap();
+            Ok(())
```

Comment:
Let's try to make this interfaces talk to each other.

```suggestion
            // This would be great
            self.metering_data_sender.send(())
            // ...otherwise
            Ok(self.metering_data_sender.send(())?)
```

Reply:
This is a testcase. There is not meaningful value in `NetworkError` corresponding to `mpsc` send errors and I do not want to add them just for a testcase. I think it is fine to `panic!` the testcase on send error.